pkg/sshd/lock.go

```diff
@@ -0,0 +1,68 @@
+package sshd
+
+import (
+	"fmt"
+	"sync"
+)
+
+// RepositoryLock interface that allows the creation of a lock associated
+// with a repository name to avoid simultaneous git operations
+type RepositoryLock interface {
+	// Lock aquires a lock for a repository
+	Lock(repoName string) error
```

Comment:
Also, @aledbf - if / when we do decide to use a timeout greater than 0, we could still provide a progress indicator in the console - I wouldn't be opposed to that.